[Test] add a simple unit tests.

nadavrot opened the pull request:
[Test] add a simple unit tests.

I plan to add new features, such as quantization, to this test in future commits.

rdzhabarov commented on tests/unittests/MLTest.cpp:
```diff
@@ -865,6 +865,89 @@ TEST_P(MLTest, convNetForImageRecognition) {
+  auto L = labels.getHandle<>();
+  auto image = images.getHandle<>();
+  unsigned numSamples = images.dims()[0];
+  image.clear(-0.5);
```
not super clear why -0.5 chosen. Is that somewhat random?

rdzhabarov commented on tests/unittests/MLTest.cpp:
```diff
@@ -865,6 +865,89 @@ TEST_P(MLTest, convNetForImageRecognition) {
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 2}, "coordinates",
+                         VisibilityKind::Public, Variable::TrainKind::None);
+
+  auto *FC0 = F->createFullyConnected("fc", input, 2);
```
is that intentional that there are no hidden layers for the network? (i assume it'd learn better with more layers and we could make that 90% check tighter).

rdzhabarov commented on tests/unittests/MLTest.cpp:
```diff
@@ -865,6 +865,89 @@ TEST_P(MLTest, convNetForImageRecognition) {
+  Function *TF = glow::differentiate(F, EE_.getConfig());
+  EE_.compile(CompilationMode::Train, TF);
+
+  Tensor images(ElemKind::FloatTy, {numSamples, 10, 10, 1});
```
nit: trainImages and testImages for the images2

nadavrot commented:
@rdzhabarov I updated the code:
1. Added another layer. 
2. Renamed the variables.
3. Changed -0.5 to zero. 

I kept the 90% hit rate because to get better results we need to train longer, and this makes the unit test run longer.
